pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,441 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+import "k8s.io/api/core/v1/generated.proto";
+import "k8s.io/api/autoscaling/v1/generated.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/gogo/protobuf/protobuf/google/protobuf/wrappers.proto";
+
+message IstioInstaller {
+    InstallerSpec spec = 3;
+    InstallerStatus status = 4;
+}
+
+// InstallerSpec defines the desired state of IstioInstaller.
+message InstallerSpec {
+    // Prefix added to all namespaces for any installed component.
+    string default_namespace_prefix = 5;
+
+    // Selection and configuration of core Istio features.
+    TrafficManagementFeatureSpec traffic_management = 25;
+    PolicyTelemetryFeatureSpec policy_telemetry = 26;
+    SecurityFeatureSpec security = 28;
+    ConfigManagementFeatureSpec config_management = 29;
```

Comment:
is this something we need to expose to users?  if galley is required for everything else to run and operate, what is the need to configure this?

Reply:
afaik galley is optional. This is an override API so there's no requirement to set any of these components if the defaults work.

Reply:
@ostromart, I agree with @linsun, if we're promoting a biased install (which I think we should, 80/20 rule), then mcp should be default, which means using galley.  That said, there should be a place to allow for whatever customization might be included in the 80%.

Reply:
This schema is both for the base spec and override spec. We're free to put whatever defaults we want into the base spec and assign base specs to profiles. 
Users would use the same schema to optionally override the base spec values. We can (and do) make galley/MCP on by default in most profiles so unless user explicitly turns it off it will be there.

Reply:
I guess the question is: should we allow users to disable mcp in a k8s installation?  Unless the answer is, yes, most users disable it, then I think the only thing we need this for is to allow users to set "operationalConfig" for galley.  If they need to disable mcp/galley, then they use the "harder" mechanism of overriding it via raw values.yaml.

Reply:
It can be done through values.yaml, in fact most of these top level API knobs have corresponding fields in values.yaml.
But, we want to give users the new API and encourage them to use it instead of values.yaml, since that API will be deprecated. 
Yes, currently we do want to allow users to disable MCP and not use galley.